sympy/geometry/util.py

```diff
@@ -310,6 +312,175 @@ def _orientation(p, q, r):
     return Polygon(*convexHull)
 
 
+def closest_points(*args):
+    """The closest pair of points among a given set of points in 2-d plane.
+
+    Parameters
+    ==========
+
+    args : a collection of Points on 2-d plane.
+
+    Returns
+    =======
+
+    pair : A pair of closest points
+
+    Notes
+    =====
+
+    This can only be performed on a set of non-symbolic points.
+
+    References
+    ==========
+
+    [1] http://www.cs.mcgill.ca/~cs251/ClosestPair/ClosestPairPS.html
+
+    [2] Sweep line algorithm
+    https://en.wikipedia.org/wiki/Sweep_line_algorithm
+
+    Examples
+    ========
+
+    >>> from sympy.geometry.util import closest_points
+    >>> points = [(5, 2), (7, 6), (3, 9), (9, 4), (8, 5)]
+    >>> closest_points(*points)
+    ((8, 5), (7, 6))
+
+    >>> from sympy.geometry.util import closest_points
+    >>> points = [(80, 89), (11, 85), (32, 51), (59, 59), (39, 80), (62, 73)]
+    >>> closest_points(*points)
+    ((62, 73), (59, 59))
+
+    """
+    from .point import Point
+    from collections import deque
+    import numbers
+
+    points = list(args)
+    if len(points) < 2:
+        raise NotImplementedError('At-least 2 points must be inserted')
+    # make sure all our points are of the same dimension
+    for i in range(0, len(points)):
+        if isinstance(points[i][0], (int, float)) == False or isinstance(points[i][1], (int, float)) == False or len(points[i]) != 2:
```

Comment:
After you use Point2D, for `points = [Point2D(1, 2), Point2D(3, 4), Point2D(5, 8), .....]` you could use something like:
```python
for point in points:
     if not all([point.x.is_number, point.y.is_number]):
         raise NotImplementedError('The input must be a set of 2D points')
```